### PR TITLE
fix: swipe between child and parent TabNavigator

### PR DESCRIPTION
**Pull request by computerjazz**
- Cancels Gesture when swiping past either end of routes, which allows parent `PanGestureHandler` to become the gesture handler.
- Provides context-based mechanism for child `PanGestureHandler`s to pass their refs up to the parent in order to use in `simultaneousHandlers` and `waitFor` props.

### Motivation
Fixes https://github.com/react-native-community/react-native-tab-view/issues/880

### Test plan

- Create nested tab navigator. Verify that swipes on the child, swipes on the parent, and swipes between child and parent all behave as expected.

panfix

### CODE REVIEW

**Comment by Gokaysim** [comment]
unfortunately, I had some issue. I have added PR with yarn add then it fails when app fetches file

/react-native-tab-view/package.json` was successfully found. However, this package itself specifies a `main` module field that could not be resolved (`/react-native-tab-view/lib/module/index.js`. Indeed, none of these files exist:

**Comment by computerjazz** [comment]
I think you need to build first, which generates `lib`.  
https://github.com/react-native-community/react-native-tab-view/blob/master/package.json#L20

**Comment by computerjazz** [comment]
@satya164 any updates on this? I've been patching this library with these changes in order for nested tab swiping to work.

**Comment by theisgroenbech** [comment]
I would love to have this merged :)

**Comment by neverlist** [comment]
I need this

**Comment by neverlist** [comment]
I copied this page,But it doesn't work， 
it show ‘Animated events must have event data’
I'm sorry, my English is not very good，I hope you understand what I mean

File: src/Pager.tsx [hunk 1]
```diff
@@ -458,12 +507,43 @@ export default class Pager<T extends Route> extends React.Component<Props<T>> {
...
   );
 
+  private toggleEnabled = () => {
+    if (this.state.enabled)
+      this.setState({ enabled: false }, () => {
+        this.setState({ enabled: true });
+      });
+  };
+
+  // Cancel gesture if swiping back from initial tab or forward from last tab.
```
Comment (osdnk) [inline src/Pager.tsx]:
```suggestion
  // Cancel gesture if swiping back from the initial tab or forward from the last tab.
```

File: src/Pager.tsx [hunk 2]
```diff
@@ -458,12 +507,43 @@ export default class Pager<T extends Route> extends React.Component<Props<T>> {
...
+  // Enables parent Pager to pick up the gesture, if one exists.
```
Comment (osdnk) [inline src/Pager.tsx]:
```suggestion
  // Enables parent Pager to pick up the gesture if one exists.
```

**Pull request merged**